extra_data/reader.py

```diff
@@ -1004,52 +1355,84 @@ def keys_detail(s, keys, prefix=''):
                 else:
                     entry_info = ""
                 dt = self.get_dtype(s, k)
-                print(f"{prefix}{k}\t[{dt}{entry_info}]")
+
+                if k in srckey_aliases[s]:
+                    alias_str = ' <' + ', '.join(srckey_aliases[s][k]) + '>'
+                else:
+                    alias_str = ''
+
+                print(f"{prefix}{k}{alias_str}\t[{dt}{entry_info}]")
 
         non_detector_inst_srcs = self.instrument_sources - self.detector_sources
         print(len(non_detector_inst_srcs), 'instrument sources (excluding detectors):')
         for s in sorted(non_detector_inst_srcs):
-            print('  -', s)
-            if not any(p.match(s) for p in details_sources_re):
-                continue
+            print('  -', s, src_alias_list(s))
 
-            # Detail for instrument sources:
+            # Detail or only aliases for instrument sources:
+            is_detailed = any(p.match(s) for p in details_sources_re)
             for group, keys in groupby(sorted(self.keys_for_source(s)),
                                        key=lambda k: k.split('.')[0]):
-                print(f'    - {group}:')
-                keys = list(keys)
-                src_data_detail(s, keys, prefix='      ')
-                keys_detail(s, keys, prefix='      - ')
+                if is_detailed:
+                    print(f'    - {group}:')
+                    keys = list(keys)
+                    src_data_detail(s, keys, prefix='      ')
+                    keys_detail(s, keys, prefix='      - ')
+                else:
+                    keys_aliases(s, keys, prefix='      - ')
 
 
         print()
         print(len(self.control_sources), 'control sources:')
         for s in sorted(self.control_sources):
-            print('  -', s)
-            if any(p.match(s) for p in details_sources_re):
-                # Detail for control sources: list keys
-                ctrl_keys = self[s].keys(inc_timestamps=False)
+            is_detailed = any(p.match(s) for p in details_sources_re)
+            has_aliases = s in srckey_aliases
+
+            print('  -', s, src_alias_list(s))
+
+            # Detail for control sources: list keys
+            ctrl_keys = self[s].keys(inc_timestamps=False)
```

Comment:
Part of why we don't already show keys by default is that it can be significantly slower to get them - we cache source names but not key names. Virtual overview files help with that, but we're still some way from being able to assume voview files exist for all runs.

So I'd like to continue ensuring that `run.info()` and `lsxfel` don't load keys for all sources. I appreciate this code is getting pretty complex - maybe it should be split out in its own module.

Reply:
Pondering this, I don't see any other way than including the aliases only when a source is detailed anyway.

Reply:
That solution would be fine, IMO. If you think it's a substantially better UX to show aliased keys even without requesting the detailed info, we could build a dict of `key_aliases_by_source` and display info for those keys. Hopefully this would still be reasonably quick, but getting the full list of keys for every source definitely isn't always.

Reply:
Okay, I reverted any key access if the source is not included in the details pattern.